pkg/platform/runtime/setup/setup.go

```diff
@@ -313,21 +349,34 @@ func (s *Setup) setupArtifact(buildEngine model.BuildEngine, a artifact.Artifact
 
 	unarchiver := as.Unarchiver()
 	archivePath := filepath.Join(targetDir, a.String()+unarchiver.Ext())
-	if err := s.downloadArtifact(unsignedURI, archivePath); err != nil {
-		return errs.Wrap(err, "Could not download artifact %s", unsignedURI)
+	downloadProgress := events.NewIncrementalProgress(s.msgHandler, events.Download, a, artifactName)
+	if err := s.downloadArtifact(unsignedURI, archivePath, downloadProgress); err != nil {
+		err := errs.Wrap(err, "Could not download artifact %s", unsignedURI)
+		s.msgHandler.ArtifactStepFailed(events.Download, a, err.Error())
+		return err
 	}
-	s.msgHandler.ArtifactDownloadCompleted(a)
+	s.msgHandler.ArtifactStepCompleted(events.Download, a)
```

Comment:
Why are we still calling it a msgHandler here? Should be `s.events...` I would think.

Reply:
Good point.  `events` seems good.